pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -63,8 +65,9 @@ const (
 )
 
 var (
-	logger       = log.Log.V(logging.DebugLevel).WithName("controller").WithValues("kind", "ConstraintTemplate", logging.Process, "constraint_template_controller")
-	discoveryErr *apiutil.ErrResourceDiscoveryFailed
+	logger         = log.Log.V(logging.DebugLevel).WithName("controller").WithValues("kind", "ConstraintTemplate", logging.Process, "constraint_template_controller")
+	discoveryErr   *apiutil.ErrResourceDiscoveryFailed
+	vapEnforcement = flag.Bool("vap-enforcement", false, "control VAP resource generation. Allowed values are false: do not generate unless generateVAP: true is added to constraint template explicitly, true: generate unless generateVAP: false is added to constraint template explicitly.")
```

Comment:
Same comment with this flag, we should explain more clearly that it only governs the default behavior for generating VAP definitions from constraint templates, right now the wording suggests the flag does more than it actually does.

Reply:
How about, `create-vap-for-templates`?